### PR TITLE
Wip/hughsie/size

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@rossburton FYI

File: data/meson.build [hunk 1]
```diff
@@ -2,7 +2,10 @@ subdir('builder')
 subdir('pki')
 subdir('remotes.d')
 subdir('bash-completion')
+
+if get_option('fish_completion')
```
Comment (superm1) [inline data/meson.build:6]: I think this should group with bash completion. You don't want any completions for an embedded target probably.

File: meson_options.txt [hunk 2]
```diff
@@ -47,3 +47,6 @@ option('curl', type : 'boolean', value : true, description : 'enable libcurl sup
 option('udevdir', type: 'string', value: '', description: 'Directory for udev rules')
 option('efi_os_dir', type: 'string', description : 'the hardcoded name of OS directory in ESP, e.g. fedora')
 option('efi_binary', type: 'boolean', value : true, description : 'generate uefi binary if missing')
+option('metainfo', type: 'boolean', value : true, description : 'install the project metainfo.xml information')
+option('fish_completion', type: 'boolean', value : true, description : 'install the fish integration files')
+option('offline', type: 'boolean', value : true, description : 'enable installing offline firmware')
```
Comment (superm1) [inline meson_options.txt:52]: This wording will be very confusing. Offline doesn't have the obvious meaning here

File: src/fu-engine.c [hunk 3]
```diff
@@ -2423,6 +2423,15 @@ fu_engine_schedule_update(FuEngine *self,
 	g_autoptr(FuHistory) history = NULL;
 	g_autoptr(GFile) file = NULL;
 
+#ifndef HAVE_FWUPDOFFLINE
```
Comment (superm1) [inline src/fu-engine.c:2426]: Maybe we should change the default in packaging we have in tree for non-x86 too. Offline is only used for EC on an x86 design right?

File: data/meson.build (hunk 1 above)
Comment (hughsie) [reply data/meson.build:6]: maybe if `get_option('completion')` instead? Or add a `bash_completion` one too?

File: meson_options.txt (hunk 2 above)
Comment (hughsie) [reply meson_options.txt:52]: `enable installing firmware using a pre-boot systemd target` perhaps?

File: data/meson.build (hunk 1 above)
Comment (superm1) [reply data/meson.build:6]: I say the former

File: meson_options.txt (hunk 2 above)
Comment (superm1) [reply meson_options.txt:52]: Yes

File: src/fu-engine.c (hunk 3 above)
Comment (hughsie) [reply src/fu-engine.c:2426]: Kinda, it's not exactly designed for x86 -- but we know on some platforms (that happen to be non-x86) that we're not going to use it.

File: src/meson.build [hunk 4]
```diff
@@ -107,7 +107,7 @@ meson.add_install_script(
       join_paths(get_option('prefix'), get_option('bindir'), 'fwupdagent')))
 endif
 
-if get_option('systemd')
+if get_option('systemd') and get_option('offline')
```
Comment (superm1) [inline src/meson.build:110]: I think there's a new matrix here that you need to check where offline is enabled but systemd is disabled. That shouldn't be a valid configuration.

File: data/meson.build (hunk 1 above)
Comment (hughsie) [reply data/meson.build:6]: Although, there seems to be some precedent for `-Dbash_completion=false` looking at the meta-openembedded tree.

File: src/meson.build (hunk 4 above)
Comment (hughsie) [reply src/meson.build:110]: So if offline is true and systemd is false do a error() in the toplevel meson.build perhaps?

Comment (superm1) [reply src/meson.build:110]: Yeah that's what I was thinking

File: meson_options.txt (hunk 2 above)
Comment (hughsie) [reply meson_options.txt:52]: Fixed.

File: src/meson.build (hunk 4 above)
Comment (hughsie) [reply src/meson.build:110]: Done.

File: data/meson.build [hunk 5]
```diff
@@ -26,13 +29,14 @@ if build_standalone
...
 endif
 
-install_data(['org.freedesktop.fwupd.metainfo.xml'],
-  install_dir: join_paths(datadir, 'metainfo')
-)
-
-install_data(['org.freedesktop.fwupd.svg'],
-  install_dir : join_paths(datadir, 'icons', 'hicolor', 'scalable', 'apps')
-)
+if get_option('metainfo')
```
Comment (superm1) [inline data/meson.build:32]: we can probably exclude this from windows build too now that this option exists

Comment (hughsie) [reply data/meson.build:32]: Yes, and FreeBSD perhaps too?

Comment (hughsie) [reply data/meson.build:32]: No, FreeBSD makes sense to keep. Ignore me.

Comment (hughsie) [reply data/meson.build:32]: Done

File: data/meson.build (hunk 1 above)
Comment (hughsie) [reply data/meson.build:6]: I've added the latter to match the others.

**Review by superm1 — DISMISSED** [review]

**Pull request merged**